src/tasks/publish/index.ts

```diff
@@ -0,0 +1,226 @@
+  .setAction(async (params, bre: BuidlerRuntimeEnvironment) => {
+    // Do param type verification here and call publishTask with clean params
+    await publishTask(
+      {
+        bumpOrVersion: params.bump,
+        existingContractAddress: params.contract,
+        managerAddress: params.managerAddress,
+        ipfsProvider: params.ipfsProvider,
+        onlyContent: params.onlyContent,
+        noVerify: params.noVerify
+      },
+      bre
+    )
+  })
+
+async function publishTask(
+  {
+    bumpOrVersion,
+    existingContractAddress,
+    managerAddress,
+    ipfsProvider,
+    onlyContent,
+    noVerify
+  }: {
+    bumpOrVersion: string
+    existingContractAddress: string
+    managerAddress: string
+    ipfsProvider: string
+    onlyContent: boolean
+    noVerify: boolean
+  },
+  bre: BuidlerRuntimeEnvironment
+): Promise<void> {
+  const aragonConfig = bre.config.aragon as AragonConfig
+  const appSrcPath = aragonConfig.appSrcPath as string
+  const distPath = aragonConfig.appBuildOutputPath as string
+  const ipfsIgnore = ['subfolder/to/ignore/**'] // Hardcoded until a better way to deal with dynamic ENS address is found
+  const selectedNetwork = bre.network.name
+  // ### Todo: Add logic to guess app name from file?
+  const appName =
+    typeof aragonConfig.appName === 'string'
+      ? aragonConfig.appName
+      : typeof aragonConfig.appName === 'object'
+      ? aragonConfig.appName[selectedNetwork]
+      : undefined
+  if (!appName)
+    throw new BuidlerPluginError(`appName must be defined in buidler.config`)
+  const contractName = getMainContractName()
+
+  // Initialize clients
+  const networkConfig = bre.network.config as HttpNetworkConfig
+  const provider = new ethers.providers.Web3Provider(
+    bre.web3.currentProvider,
+    networkConfig.ensAddress && {
+      name: bre.network.name,
+      chainId: networkConfig.chainId || 5555,
+      ensAddress: networkConfig.ensAddress
+    }
+  )
+
+  const prevVersion = await _getLastestVersionIfExists(appName, provider)
+
+  const { bump, nextVersion } = parseAndValidateBumpOrVersion(
+    bumpOrVersion,
+    prevVersion ? prevVersion.version : undefined
+  )
+  logMain(`Applying version bump ${bump}, next version: ${nextVersion}`)
+
+  // Using let + if {} block instead of a ternary operator
+  // to assign value and log status to console
+  let contractAddress: string
+  if (onlyContent) {
+    logMain('No contract used for this version')
+    contractAddress = zeroAddress
+  } else if (existingContractAddress) {
+    logMain('Using provided contract address')
+    contractAddress = existingContractAddress
+  } else if (!prevVersion || bump === 'major') {
+    logMain('Deploying new implementation contract')
+    contractAddress = await _deployMainContract(contractName, noVerify, bre)
+  } else {
+    logMain('Reusing contract from previous version')
+    contractAddress = prevVersion.contractAddress
+  }
+  logMain(`contractAddress: ${contractAddress}`)
```

Comment:
Nitpick: I think it'd be nicer to log this inline with the log describing the contract selected